comp/core/autodiscovery/autodiscoveryimpl/autoconfig.go

```diff
@@ -64,31 +110,40 @@ func (l *listenerCandidate) try() (listeners.ServiceListener, error) {
 	return l.factory(l.config)
 }
 
-// NewAutoConfig creates an AutoConfig instance and starts it.
-func NewAutoConfig(scheduler *scheduler.MetaScheduler, secretResolver secrets.Component) *AutoConfig {
-	ac := NewAutoConfigNoStart(scheduler, secretResolver)
-
-	// We need to listen to the service channels before anything is sent to them
-	go ac.serviceListening()
-
+// newAutoConfig creates an AutoConfig instance and starts it.
+func newAutoConfig(deps dependencies) autodiscovery.Component {
+	ac := NewAutoConfigNoStart(scheduler.NewMetaScheduler(), deps.Secrets)
```

Comment:
Do we have to include the `NoStart` as part of the function name? Since we do not have an alternative function that do start the autodiscovery component we might as well just have `NewAutoConfig`

Also, those the function has to be public? I think the desired pattern is to have the initialization function of the components to be private

Reply:
Updated and use NewAutoConfig now. NewAutoConfig is public because it is used by two unit tests in other pkg. In these cases, dependency on tagger component is not needed and start function is not called either.